README.md

## <p align="center">![Screenshot](https://nr1.nu/f/sqqmu.gif)


### News 1: 

- XKCD has been hacked, over 500 000 accounts is exposed.

Keep emagnet running 24/7 and you will get the leaked accounts, sooner or later they will popup on pastebin. Emagnet will get them all for you with 100% _guarantee_!
![Screenshot](https://1.bp.blogspot.com/-ko993L8-Opo/XW6HHkV3k1I/AAAAAAAA07U/VNWUNRHXZbwEnb1S6ewJ6JYxPCgiy4qiQCLcBGAs/s728-e100/xkcd-hacked.png)
![Screenshot](https://i.imgur.com/mTsuqNZ.png)

## News 2:

 Latest leak that has been added on HaveIBeenPwned is 'Void To' 
 But I doubt they really have the leaked db since I searched for several email addresses from the link below and they was not reported as leaked on haveibeenpwned, troy hunt sucks, he have done nothing, really nothing but he take all creds, urgh! Just saying ;).. All creds goes to Srkr ТЫ С4АСТЛN:

Go grab the leaked void to leaked db if you are interested before they will remove it (probably): https://mega.nz/#F!EnxggCiY!btfevmu3fO4AQrWe-81z6A!hm50kRBJ

<a href="https://github.com/wuseman/EMAGNET"><img src="https://img.shields.io/github/languages/top/wuseman/emagnet.svg?color=magenta&label=Bash%2FShell"></a><a href="https://github.com/wuseman/EMAGNET/issues?q=is%3Aissue+is%3Aclosed">
<img src="https://img.shields.io/github/issues-closed/wuseman/emagnet.svg?color=light&label=Closed%20Issues"></a>
 <a href="https://github.com/wuseman/EMAGNET/issues"><img src="https://img.shields.io/github/issues-raw/wuseman/emagnet.svg?color=orange&label=Open%20Issues"></a><img src="https://img.shields.io/github/last-commit/wuseman/emagnet.svg?color=darkmagenta&label=Latest%20Commit"><a href="https://twitter.com/wuseman1">
 <img src="https://img.shields.io/website/https/nr1.nu.svg?down_color=darkred&down_message=DOWN&label=Nr1.nu%2Femagnet&up_message=UP"><img src="https://img.shields.io/github/license/wuseman/emagnet.svg?color=blue&label=License"></a></a></a>
</a>
</p>

| Current Version    |  Released      | News                            | Tested On                          |
| :----------------- | :------------- | :-------------------------------- | :----------------------------------|
| `3.4`              |      -:-       |  Full support on android devices, no root required    | Linux                               |


Emagnet is a very powerful tool for it's purpose wich is to capture  email addresses and passwords from leaked databases uploaded on pastebin. It's almost impossible to find leaked passwords when they are out of list on pastebin.com. Either they have been deleted by pastebin's techs or the uploads is just one in the crowd. To be honest it's easier to find a needle in a haystack then find outdated uploads on pastebin with the data we want to collect. 

### Known Issues That Will Be Fixed In Next Push!

- [x] SSH Bruterforcer isn't completely working anymore 
- [ ] Requirements packages for android devices when using termux app isnt working with brute force tools + screen
- [ ] Sending wrong data into emagnet.log for wich files that contains correct data 
- [ ] Re-add the feature that bypassing captcha when pastebin asking for accept captcha
- [ ] Prio 1. Fix so we don't need visit pastebin more then once for check for bans and captcha..

### TIPS:

##### Bruteforce on your own risk.

For people that like things sorted. If you have some big archives with leaked databases stored (or small) on your harddrive but you are to lazy to figure out wich works and not on sites like spotify or gmail for example, then just create a a random text file in "$HOME/.config/emagnet/tmp/some-random-file" - Add all logins as: mail:password one per line - Now just run emagnet as usual and you will bruteforce all accounts you added to the file and all cracked accounts will be stored in log dirs! Once done, you will have a good list for accounts that actually works for real or at least you know wich accounts you can put at top of your text file that you know is legit.

Don't forget to use vpn, tor or just edit curl command if you want to use a socks5 proxy / ssh tunnel: For a socks5 proxy use: curl --socks5-hostname localhost:port ...... and for tor use curl --socks5 127.0.0.1:9150 ......

### Demo - How it works
#### Left side is how you see it - Right side is what actually happening in the background
![Screenshot](.preview/emagnet2-debug.gif)

### Get Started On Linux/MacOSX

    git clone https://github.com/wuseman/emagnet
    cd emagnet
    chmod +x emagnet*
    .
    ./emagnet --emagnet
    
### Get started on Android via Termux

See video [here](https://nr1.nu/android-emagnet.gif)

* Download termux from play store [here](https://play.google.com/store/apps/details?id=com.termux&hl=en_US)

Once installed, copy and paste below commands: 

     echo -e '
     pkg update
     pkg upgrade -y
     pkg install wget curl git -y
     git clone https://github.com/wuseman/emagnet

     cd emagnet
     bash emagnet --emagnet' > emagnet-termux.sh
     bash emagnet-termux.sh
    
###  Get Started On Windows 10

Please visit my installation [wiki](https://github.com/wuseman/EMAGNET/wiki/Configurations-&-Installation) for more info (includes a video)

### Wiki Sections:

- [About](https://github.com/wuseman/EMAGNET/wiki/ABOUT) - 
_How everything started._

- [Previews](https://github.com/wuseman/EMAGNET/wiki/PREVIEWS) - 
_Previews can be found here._

- [Faq](https://github.com/wuseman/EMAGNET/wiki/FAQ) - 
_How to grab your visa card if it has been leaked. Also get answers why we not using TOR._

- [Configurations-&-Installation](https://github.com/wuseman/EMAGNET/wiki/Configurations-&-Installation) - 
_Get started with spotify brute forcing - How emagnet will work with your openvpn files._

- [Regex - Tips For Search](https://github.com/wuseman/EMAGNET/wiki/Searching-&-Regex) - How To Find your facebook credenticals, if it has been leaked._

### System requirements

- Bash     - Find more info about _bash_ [here](https://www.gnu.org/software/bash/)
- Wget     - Find more info about _wget_ [here](https://www.gnu.org/software/wget/)
- Curl     - Find more info about _curl_ [here](https://github.com/curl/curl)

#### BBC NEWS: ["Pastebin: Running the site where hackers publicise their attacks"](https://www.bbc.com/news/technology-17524822) 
- Emagnet is your best friend for get the leaks

## Changelog

[Versions changelog](CHANGELOG.md).

## Authors: 

* **wuseman <wuseman@nr1.nu\>** 

## License

This project is licensed under the GNU General Public License v3.0 - see the [LICENSE.md](LICENSE.md) file for details

### Contact

  If you have problems, questions, ideas or suggestions please contact me on *_wuseman@nr1.nu_  - For faster contact visit freenode irc network or the webchat and type '/msg wuseman hi!' in the input bar and I will reply you ASAP I will see the message.
  
  Enter Freenodes network via your own client 'chat.freenode.com:+6697 or use their new web client [here](https://webchat.freenode.net/)

### Notice

Attacking different kinds of accounts via emagnet that you have not been granted or allowed to attack is strictly prohibited and it breaks the law. The punishment is hard and you can even get into prison in some countries just for trying to attack for intrusion. With this said, it's *important* that all users is aware of this and when you have cloned or downloaded it's fully up to every user to take responsibility over their own actions. wuseman cannot be held responsible for the actions of any user, all users using Emagnet on their own responsibility. 

Developer: "All my previews where a brute force attack has been done is under controlling forms with 100% fully permissions by the owners. If you have any questions about this then you are welcome to contact me or the owner."

### Haters Gonna Hate

If you are one of these who dislikes _EMAGNET_ and believe the program has been developed for a reason that would break the law then I am not interested in taking part of your opinions, keep them for _yourself_! Emagnet does **NOT** leak any data at all either to the developer(s) or anyone else. No statistics at all to track any user so if you want to contact me for ask who it might was who downloaded emagnet a specific date is completely useless since i really have no idea, and to be honest I don't care.

Feel free to read the history about emagnet [here](https://github.com/wuseman/EMAGNET/wiki/About) and how everything started about this project.

#### Development of emagnet is active and is updated frequently, please use the latest version if you report issues/bugs.

# Greetings: 

_m1st_ that deliver legit leaks for us daily.

And to all ppl that is trying to sell public leaks and steal the real hackers job, f*ck you! This is one reason why I started this project, I hope this project will get widely spreaded so you will earn 0.00$ on your re-edited malware shit!

Cheers!

### Emagnet is a private project since 2015 and was released in June @ 2018, to be continued. 





CHANGELOG.md
### CHANGELOG FOR EMAGNET

##### EMAGNET IS A NEW PROJECT, I WILL RELEASE VERSION IN SHORT INTERVALLS SO I CAN GO BACK TO SEE EVERYTHING I ADDED AND REMOVED DURING THE TIME

# DESCRIPTION

+ = NEW FEATURE
- = DELETED FEATURE
r = RE-ADDED FEATURE FROM AN EARLIER VERSION
     If you have been using emagnet since version 1.2 you know some stuff have been added and removed during the time frequently, these features
     that has been re-added is now stable and not "half done" so they are here for stay.
R = REBUILT


# Version 3.4

+    Fixed so if we are busted and being asked for captcha then we changing IP
+    Set wich file you want to send to all email addresses by: -s /path/to/textfile.txt
+    Set vpn provider by: -v -p <provider>
+    Support for instagram brute forcing
+    Changed layout, removed spaces between everything we found
+    Re-added ssh bruteforcer, nothing wont be stored to log, read script description for why we don't 
+    Small and general fixes for a better setup
+    Better regex and added it easier to read how we grab stuff
+    Better quoting
+    Renamed all functions to emagnet_functionname
+    Added a more elegant way to write analyzing.
+    Added HEADER to emagnet.conf, default is Mosaic (first header ever made)
+    Added LIBSPOITFY to emagnet.conf
+    Added emagnet.conf into root dir
+    Added support for merge logs in to archive
+    Added requirements for vpn part (openvpn)
+    Edited how we read LICENSE
+    Added a function for colors so we can use a more elegant way to print stuff

-    Removed color function again since it wont work on android devices
-    Removed .emagnetconf/emagnet.conf and .emagnetconf/emagnet.conf.bak
-    Removed requirements for a browser to be installed
-    Remove temp files so they wont be stored in tmp folder 
-    Removed spammer
-    Removed stats for rebuilding
-    Removed safety part that added to log if someone otherwise then root tried to run script (useless since long time ago)
-    Removed setup script
-    Removed elinks and lynx requirements so we can use emagnet on more systems without install more tools then we really need
-    Removed requirements part as default, it will be used when you using stuff that may need requirements like smtp/parallel for spammer & ssh bruteforce

# Version 3.2

-    Removed fuzzer, and find temporary
+    Spotify BRUTEFORCE has been added
-    Removed requirements for netcat since we dont have any use for netcat anymore
+    emagnet-setup.sh has now been moved to emagnet
+    Notifications has been added, this require notify-send
+    Toggle between true or false via --notify
+    SSH bruteforce has been added
+    Removed almost all comments since it toke alot of place
+    General fixes

# Version 3.1

+    From now you will see how many files the current session was downloading
+    Fixed bug so emagnet-files never was created and no email files was stored, they was still stored in logs so nothing was missed
+    Fixed a new layout on emagnet stats
+    Fixed so we wont download duplicates, this have been tricky during time as you might have noticed but finally it is working 100% so no more dupes will be downloaded, you will see this
     thanks to the new "Files has been downloaded" feature when you run emagnet
+    Fixed a new layout for email bruter
+    Fixed new layout for help
+    Fixed so unneccessary stuff wont be sourced or loaded while we run --help for example, this will print help only (will even be possible without root)
+    General fixes..
+    Added author for print author information
+    Added license for print license information
+    Added setup for re-run setup (in next version emagnet-setup will be built in, hence this was added since it's useless otherwise since you can run emagnet-setup.sh otherwise ;) )
+    Added emagnet backup for backup all, archive or incoming folder
+    Added long + short options for everything, removed while getopts..
+    Added so you can toggle between false or true on vpn
+    Added so you can set time with --time 5 instead of changing conf file all the time
+    Better support for VPN, it will try connect again if it didnt connect the first time, the third time you will be asked to fix it manually since something is probably wrong?
+    If no new files has been downloaded you will see that 0 files has been downloading by blinking + you will be noticed that you should raise the REFRESH time probably? (this happends mostly if you run emagnet within 3 seconds or less for testing
+    Pastebin has added a new protection, if you are scraping pastebin to fast you will be banned from archive page and will only be available to get 8 downloads at each visit - Emagnet will notice when this happends from now.

# Version 3.0

-    Removed LOGS2 from emagnet.conf
R    Spammer has been rebuilded totally since it didnt work any good before. From now we gonna spam all targets from our daily dir if there is any email addresses stored
r    Added inotify script example to README again
+    Added BACKUPS to emagnet.conf
+    Cleaned up alot and changed order on all functions so it's easier to understand how emagnet is built
+    Added better description in configuration file
+    Added NOTIFY to emagnet.conf so we can get notification if we are using notify-send (Linux Only)
+    Added VPN settings in /etc/emagnet.conf, if we getting banned we can choose to get a new IP from our vpn provider instead of idle 3600+ seconds so we wont miss anything ;-)
+    Added SSH settings in /etc/emagnet.conf for ssh bruteforcer (COMMING SOON)

emagnet-setup.sh
#!/bin/bash
################################################################################
################################################################################
####                                                                       #####
#### A notice to all nerds.                                                #####
#### If you will copy developers real work it will not make you a hacker.  #####
#### Resepect all developers, we doing this because it's fun!              #####
####                                                                       #####
################################################################################
################################ SOURCE CODE ###################################
################################################################################
###################### EMAGNET WAS FOUNDED BY WUSEMAN ##########################
################################################################################
####                                                                       #####
####   Emagnet - Tool for find leaked databases, email-addresses, passwords#####
####   credit cards, paypal accounts, darknet urls and much much more      #####
####   from uploaded files on pastebin.                                    #####
####                                                                       #####
####  Copyright (C) 2018-2030, wuseman                                     #####
####                                                                       #####
####  This program is free software; you can redistribute it and/or modify #####
####  it under the terms of the GNU General Public License as published by #####
####  the Free Software Foundation; either version 2 of the License, or    #####
####  (at your option) any later version.                                  #####
####                                                                       #####
####  This program is distributed in the hope that it will be useful,      #####
####  but WITHOUT ANY WARRANTY; without even the implied warranty of       #####
####  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the        #####
####  GNU General Public License for more details.                         #####
####                                                                       #####
####  You must obey the GNU General Public License. If you will modify     #####
####  emagnet file(s), you may extend this exception to your version       #####
####  of the file(s), but you are not obligated to do so.  If you do not   #####
####  wish to do so, delete this exception statement from your version.    #####
####  If you delete this exception statement from all source files in the  #####
####  program, then also delete it here.                                   #####
####                                                                       #####
####  Contact:                                                             #####
####          IRC: Freenode @ wuseman                                      #####
####          Mail: wuseman <wuseman@nr1.nu>                               #####
####                                                                       #####
################################################################################
######## The Nr1 tool for find latest leaked databases online @ 2o18 ###########
################################################################################

if [[ $EUID -gt "0" ]]; then printf "emagnet: internal error -- root privileges is required\n"; exit 0; fi
cp .emagnetconf/emagnet.conf.bak .emagnetconf/emagnet.conf
CONF=".emagnetconf/emagnet.conf ";source $CONF
SCRIPT="$(pwd)/emagnet"
LYNX="/usr/bin/lynx"
ELINKS="/usr/bin/elinks"
WGET="/usr/bin/wget"
CURL="/usr/bin/curl"
SCREEN="/usr/bin/screen"
PARALLEL="/usr/bin/parallel"
DISTRO=$(cat /etc/*release | head -n 1 | awk '{ print tolower($1) }' | cut -d= -f2)

banner() {
cat << "EOF"
     _                      _______                      _
  _dMMMb._              .adOOOOOOOOOba.              _,dMMMb_
 dP'  ~YMMb            dOOOOOOOOOOOOOOOb            aMMP~  `Yb
 V      ~"Mb          dOOOOOOOOOOOOOOOOOb          dM"~      V
          `Mb.       dOOOOOOOOOOOOOOOOOOOb       ,dM'
           `YMb._   |OOOOOOOOOOOOOOOOOOOOO|   _,dMP'
      __     `YMMM| OP'~"YOOOOOOOOOOOP"~`YO |MMMP'     __
    ,dMMMb.     ~~' OO     `YOOOOOP'     OO `~~     ,dMMMb.
 _,dP~  `YMba_      OOb      `OOO'      dOO      _aMMP'  ~Yb._
             `YMMMM\`OOOo     OOO     oOOO'/MMMMP'
     ,aa.     `~YMMb `OOOb._,dOOOb._,dOOO'dMMP~'       ,aa.
   ,dMYYMba._         `OOOOOOOOOOOOOOOOO'          _,adMYYMb.
  ,MP'   `YMMba._      OOOOOOOOOOOOOOOOO       _,adMMP'   `YM.
  MP'        ~YMMMba._ YOOOOPVVVVVYOOOOP  _,adMMMMP~       `YM
  YMb           ~YMMMM\`OOOOI`````IOOOOO'/MMMMP~           dMP
   `Mb.           `YMMMb`OOOI,,,,,IOOOO'dMMMP'           ,dM'
     `'                  `OObNNNNNdOO'                   `'
                           `~OOOOO~'

EOF
printf "%62s \n\n" | tr ' ' '='
}

alreadyconfigured() {
clear; banner
if [[ -f /etc/emagnet.conf ]]; then
    printf "It seems emagnet.conf has already been configured are you\n"
    read -p "really sure you want to overwrite /etc/emagnet.conf (y/n): " yupiamsure
 if [[ $yupiamsure = "n" ]]; then
    printf "\nAborted\n\n"
    exit
fi
    cp /etc/emagnet.conf /etc/emagnet.conf.bak
    rm /etc/emagnet.conf
    printf "\nCreated a backup file for you: \e[1m/etc/emagnet.conf.bak\n\n\e[0m"
   sleep 2
fi
}

alreadyconfigured

requirements() {
echo -e "\n\n\033[1mDEPENDENCIES SETUP:\033[0m\n\033[1m----------------------\033[0m"

# Since ubuntu have named netcat on it's own way
if [[ $DISTRO = "ubuntu" ]]; then
     NETCAT="/bin/netcat";
else
     NETCAT="/usr/bin/nc"
fi

checkwget() {
if [[ -x  $WGET ]]; then
    printf "\nDetected WGET"; printf "%46s[\e[1;32mOK\e[0m]\n"|tr ' ' '.'
else
    printf "\nDetected wget %46s[\e[1;32mNO\e[0m]\n"|tr ' ' '.' >> packages
    printf "\nDetected WGET"; printf "%46s[\e[1;31mNO\e[0m]\n"|tr ' ' '.'
fi
}

checkcurl() {
if [[ -x  $CURL ]]; then
    printf "Detected CURL"; printf "%46s[\e[1;32mOK\e[0m]\n"|tr ' ' '.'
else
    printf "Detected curl%46s[\e[1;31mNO\e[0m]\n"|tr ' ' '.' >> packages
    printf "Detected CURL"; printf "%46s[\e[1;31mNO\e[0m]\n"|tr ' ' '.'
fi
}

checkscreen() {
if [[ -x  $SCREEN ]]; then
    printf "Detected SCREEN"; printf "%44s[\e[1;32mOK\e[0m]\n"|tr ' ' '.'
else
    printf "Detected screen %44s[\e[1;31mNO\e[0m]\n" |tr ' ' '.' >> packages
    printf "Detected SCREEN"; printf "%44s[\e[1;31mNO\e[0m]\n"|tr ' ' '.'
fi
}

checkparallel() {
if [[ -x  $PARALLEL ]]; then
    printf "Detected PARALLEL"; printf "%42s[\e[1;32mOK\e[0m]\n"|tr ' ' '.'
if [[ ! -f packages ]]; then
     printf "\nAll dependencies has been found, moving on\n"
     wip;emagnethome;wgettimer;settime
     exit 1
 fi
else
    printf "Detected PARALLEL"; printf "%42s[\e[1;31mNO\e[0m]\n"|tr ' ' '.'
    printf "Detected parallel %42s[\e[1;31mNO\e[0m]\n"|tr ' ' '.' >> packages
 fi
}



checkwget;checkcurl;checkscreen;checkparallel;checknetcat

if [[ -f packages ]]; then
missed="$(awk -F. '{print $2}' packages|sed 's/ /,/g'|xargs|sed 's/ /, /g')"
missed2="$(cat packages|awk -F. '{print $2}' packages)"
missed3="$(cat packages|awk -F. '{print $2}' packages|tr ' ' '\n'|wc -l)"
if [[ $missed3 = "2" ]]; then
    missed="$(awk -F. '{print $2}' packages|xargs|sed 's/ /\ \& /g')"
    printf "\nYou must have \e[1;37m$missed\e[0m installed before emagnet can run\n\n"
    printf "Do you want to install the missed packages: (y/n): "; read missedpackages
    echo ""
    rm packages &> /dev/null
else
    printf "\nYou must have \e[1;37m$missed\e[0m installed before emagnet can run\n\n"
    printf "Do you want to install the missed packages: (y/n): ";
    read missedpackages
    echo ""
    rm packages &> /dev/null
fi

if [[ ! $missedpackages = "y" ]]; then
      echo -e "Setup can't continue until these packages has been installed, aborted...\n";exit 0;else
 case $missedpackages in
   y|yes|YES)
        if [[ $DISTRO = "gentoo" ]]; then emerge --ask  $missed2;echo ""
      elif [[ $DISTRO = "sabayon" ]]; then emerge --ask $missed2;echo ""
      elif [[ $DISTRO = "ubuntu" ]]; then apt-get install $missed2 -y; echo ""
      elif [[ $DISTRO = "kali" ]]; then apt-get install $missed2 -y; echo ""
      elif [[ $DISTRO = "debian" ]]; then apt-get install $missed2; echo ""
      elif [[ $DISTRO = "mint" ]]; then apt-get install $missed2; echo "";else
        echo "Emagnet is not supported for $DISTRO, please install $missed manually.";exit 0
        fi
        ;;

   "*") echo "Can't continue until these packages has been installed, aborted..."; exit 0 ;;
   \?)  echo "Please enter a proper answer y=yes N=no"
 esac
       fi
else
       wip;emagnethome;wgettimer;settime
       exit 0
fi
       wip;emagnethome;wgettimer;settime
       exit 0
}

wip() {
echo -e "\n\n\033[1mIP-ADDRESS SETUP:\033[0m\n\033[1m----------------------\033[0m
Set your private ip here if you don't wanna take the risk to get
your 'private' ip number banned on www.pastebin.com for around
a time limit of ~1200-1600 seconds. Remember this will happends if
you will set time limit to 59 seconds or less in next option.

If this is something you don't care about then please just hit
enter and set your ip to 0.0.0.0. If you wont do this and time
has been set to <59s emagnet will warn you about about this every
time you executing emagnet.

This setting exist so all users is aware that pastebin will ban
your ip if you will visit the page too often.
"
if [[ -d "/usr/local/bin" ]]; then cp ./emagnet /usr/local/bin &> /dev/null;fi

myip="$(curl -s https://nr1.nu/i/)"
printf "Is \e[1;1m$myip\e[0m your private IP-Address (yes/no): "; read realip
case "$realip" in
  YES)
      sed -i '40d' $CONF;sed -i '40 i MYIP=' $CONF;sed -i "s/MYIP=/MYIP=$myip/g" $CONF
      printf "\nConcfig file has been updated, your IP has been set to: \e[1;1m$myip\e[0m\n"
      ;;
  y)
      sed -i '40d' $CONF;sed -i '40 i MYIP=' $CONF;sed -i "s/MYIP=/MYIP=$myip/g" $CONF
      printf "\nConfig file has been updated, your IP has been set to: \e[1;1m$myip\e[0m\n"
      ;;
  yes)
      sed -i '40d' $CONF;sed -i '40 i MYIP=' $CONF;sed -i "s/MYIP=/MYIP=$myip/g" $CONF
      printf "\nConfig file has been updated, your IP has been set to: \e[1;1m$myip\e[0m\n"
      ;;

   *) read -p "Enter your current real IP-Adress: " currentip;sed -i "s/MYIP=/MYIP=$currentip/g" $CONF
      printf "\nConfig file has been updated, your IP has been set to: \e[1;1m$currentip\e[0m\n"
      ;;
esac
if [[ -z $realip ]]; then
      sed -i '40d' $CONF;sed -i '40 i MYIP=' $CONF;sed -i "s/MYIP=/MYIP=127.0.0.1/g" $CONF
fi
}

settime() {
while true; do echo -e "\n\n\033[1mTIME SETUP:\033[0m\n\033[1m----------------------\033[0m
By updating emagnet in 59 seconds or less you will get ip banned
by pastebin after ~1200 seconds, if you choose to set emagnet to refresh
in 59 seconds or less anyway then \e[1;1mdo\e[0m not leave emagnet running
in background by quiet mode(option -S) cause this will not result in more
files to be downloaded because you will be banned in 1200 seconds wich means
a loss of 150 uploaded files instead because  1200seconds-3600seconds = ~150 uploads

If you want to run emagnet in background without being banned and
capture files 24/7 then you MUST set emagnet to run every >59 seconds
to stay safe from get a temporary ip ban.

Whatever, no matter what time you will choose to set here emagnet
will take care of the situation. If you will be banned from pastebin
emagnet will sleep for 1200 seconds and then it will startover again.

\e[1;1mIT IS RECOMMENDED TO SET 59 SECONDS OR MORE\e[0m
"
read -p "Set refresh time in seconds for how often emagnet should download new files from pastebin
Time: (Default: 60 Seconds): " o

if [[ $o -lt "59" ]]; then
if [[ -z $o ]]; then
        sed -i '30d' $CONF
        sed -i "30 i TIME=60" $CONF
        printf "\nConfig file has updated the refresh time to 60 seconds\n"
        echo -e "\n\n\033[1mSETUP FINISHED:\033[0m\n\033[1m----------------------\033[0m"
        printf "Successfully generated /etc/emagnet.conf, have fun!\n\n"
        printf "\nNow you're all set - Have phun - \e[1;1m'bash emagnet --help' \e[1;1m:-)\e[0m\n\n"
        cp .emagnetconf/emagnet.conf  /etc/
        exit 1
else
#          printf "\nBy updating emagnet in 59 seconds or less you will get ip-banned\nunless you setup your own script that is able to control\nthe refresh time X times before changing IP..\n\n"
read -p "Are you REALLY sure you want to set $o seconds (YES/no): " rlysure
fi
 case $rlysure in
   YES)
        sed -i '30d' $CONF
        sed -i "30 i TIME=$o" $CONF
        printf "\nConfig file has updated the refresh time to $o seconds\n"
        printf "\n\e[1;31mYou have been warned, expect a ban within ~300 seconds!\n\e[0m"
        echo -e "\n\n\033[1mSETUP FINISHED:\033[0m\n\033[1m----------------------\033[0m"
        printf "Successfully generated /etc/emagnet.conf, have fun!\n\n"
        printf "\nNow you're all set - Have phun - \e[1;1m'bash emagnet --help' \e[1;1m:-)\e[0m\n\n"
        cp .emagnetconf/emagnet.conf  /etc/; break ;;

    *)  printf "\nSet a number between 10 and 3600 is recommended..\n\n"; continue ;;
 esac
fi
    sed -i "30d" $CONF
    sed -i "30i TIME=$o" $CONF
    printf "\nRefresh time has been set to: $o seconds..\n\n"
    echo -e "\n\n\033[1mSETUP FINISHED:\033[0m\n\033[1m----------------------\033[0m"
    printf "Successfully created /etc/emagnet.conf\n"
    printf "\nNow you're all set - Have phun - \e[1;1m'bash emagnet --help' \e[1;1m:-)\e[0m\n\n"
    cp .emagnetconf/emagnet.conf  /etc/
  exit
done
}

wgettimer() {
echo -e "\n\n\033[1mPROCESSOR SETUP:\033[0m\n\033[1m----------------------\033[0m
This setting will increase the speed you will download files from
pastebin dramatically, this wont affect how fast you will get banned
"
printf "Your\e[1;1m$(cat /proc/cpuinfo|grep 'model name'| awk -F: '{print $2}'|uniq)\e[0m processor has \e[1;1m$(nproc)\e[0m threads\n"
printf "Do you want emagnet to use all \e[1;1m$(nproc)\e[0m threads when downloading: (y/N): "; read threads
if [[ $threads = "y" ]]; then
    sed -i '100d' $CONF;sed -i "100 i THREADS=" $CONF;sed -i "s/THREADS=/THREADS=$(nproc)/g" $CONF;
    printf "\nConfig file has been updated, cpu threads has been set to \e[1;1m$(nproc)\e[0m thread(s)\n";else
    read -p "How many threads do you want to use (Ex: 2): " threadstouse
    sed -i '100d' $CONF;sed -i "100 i THREADS=" $CONF;sed -i "s/THREADS=/THREADS=$threadstouse/g" $CONF
    printf "\nConfig file has been updated, cpu threads has been set to \e[1;1m$threadstouse\e[0m thread(s).\n"
fi
}

emagnethome() {
echo -e "\n\n\033[1mPATH SETUP:\033[0m\n\033[1m----------------------\033[0m"

printf "Please specify in wich folder you to store all downloaded files from pastebin\n"
read -p "Path (Default: /opt/emagnet): " ehomedir
if [[ $ehomedir = "" ]]; then
  mkdir -p /opt/root
  sed -i '72d' $CONF;sed -i "72 i EMAGNET=\/opt\/emagnet" $CONF
  printf "\nConfig file has been updated, using \e[1;1m/opt/emagnet\e[0m as emagnets homedir. \n";else
  eehomedir="$(echo $ehomedir | sed 's/\/\///g')";sed -i '72d' $CONF;sed -i "72 i EMAGNET=$ehomedir" $CONF
  printf "\nConfig file has been updated, home dir has been set to: \e[1;1m$eehomedir/emagnet\e[0m\n"
fi
}

choosebrowser() {
if [[ -f $LYNX && -f $ELINKS ]]; then
echo -e "\033[1mBROWSER SETUP:\033[0m\n\033[1m----------------------\033[0m
Choose wich browser you prefer to use when emagnet will visit
pastebin.com for the latest uploads urls on site. If you don't
know which one you prefer it doesn't matter, just choose one.
"
    printf "It seems you have both lynx and elinks2 installed, you must choose one\n"; read -p "Option: (lynx/elinks): " browsertouse
if [[ $browsertouse = "lynx" ]]; then
    sed -i "181d" $CONF;sed -i '181 i BROWSER=lynx' $CONF
    printf "\nConfig file has been updated, lynx will be used for downloading files\n"
    else
    sed -i "181d" $CONF;sed -i '181 i BROWSER=elinks' $CONF
    printf "\nConfig file has been updated, elinks will be used for downloading files\n"
    fi
    elif [[ -f $LYNX && ! -f $ELINKS ]]; then
    sed -i "181d" $CONF;sed -i '181 i BROWSER=lynx' $CONF
    elif [[ ! -f $LYNX && -f $ELINKS ]]; then
    sed -i "181d" $CONF;sed -i '181 i BROWSER=elinks' $CONF
    else
    echo -e "\n\n\033[1mBROWSER SETUP:\033[0m\n\033[1m----------------------\033[0m
Choose wich browser you prefer to use when emagnet will visit
pastebin.com for the latest uploads urls on site. If you don't
know which one you prefer it doesn't matter, just choose one.
Emagnet require lynx or elinks to be installed, you must choose one"

read -p "Prefered browser to install: (lynx/elinks): " browsertouse2
 case $browsertouse2 in
   elinks)
      printf "\nGoing to install $browsertouse2, setup will continue when $browsertouse2 has been installed..\n\n"
      if [[ $DISTRO = "gentoo" ]]; then emerge --ask elinks
       requirements;wip;emagnethome;wgettimer;settime; exit 0;fi
      if [[ $DISTRO = "sabayon" ]]; then emerge --ask elinks
       requirements;wip;emagnethome;wgettimer;settime;exit 0;fi
      if [[ $DISTRO = "ubuntu" ]]; then apt-get install elinks -qq -y;
       requirements;wip;emagnethome;wgettimer;settime;exit 0;fi
      if [[ $DISTRO = "debian" ]]; then apt-get install elinks -qq -y;
       requirements;wip;emagnethome;wgettimer;settime;exit 0;fi
      if [[ $DISTRO = "mint" ]]; then apt-get install elinks;
       requirements;wip;emagnethome;wgettimer;settime;exit 0;fi
      if [[ -n $DISTRO ]]; then echo "Emagnet is not supported for $DISTRO, please install elinks manually."; exit 0; fi 
       sed -i "181d" $CONF;sed -i '181 i BROWSER=elinks' $CONF
       printf "\nConfig file has been updated, elinks will be used when downloading files from pastebin" $SCRIPT ;;
   lynx)
      printf "\nGoing to install $browsertouse2, setup will continue when $browsertouse2 has been installed..\n\n"
      if [[ $DISTRO = "gentoo" ]]; then emerge --ask lynx
       requirements;wip;emagnethome;wgettimer;settime;exit 0;fi
      if [[ $DISTRO = "sabayon" ]]; then emerge --ask lynx; 
       requirements;wip;emagnethome;wgettimer;settime;exit 0;fi
      if [[ $DISTRO = "ubuntu" ]]; then apt-get install lynx -qq -y;
       requirements;wip;emagnethome;wgettimer;settime;exit 0;fi
      if [[ $DISTRO = "debian" ]]; then apt-get install lynx -qq -y;
       requirements;wip;emagnethome;wgettimer;settime;exit 0;fi
      if [[ $DISTRO = "mint" ]]; then apt-get install lynx;
       requirements;wip;emagnethome;wgettimer;settime;exit 0;fi
      if [[ -n $DISTRO ]]; then echo "Emagnet is not supported for $DISTRO, please install lynx or elinks manually."; exit 0; fi
       sed -i "181d" $CONF;sed -i '181 i BROWSER=lynx' $CONF
       printf "\nConfig file has been updated, lynx will be used when downloading files from pastebin" $SCRIPT ;;
   N) exit 0 ;;
   \?) echo "Please enter a proper answer y=yes N=no" ;;
 esac
fi
}


clear;banner;choosebrowser;requirements;wip;emagnethome;wgettimer;settime

